server/odc-service/src/main/java/com/oceanbase/odc/service/task/service/StdTaskFrameworkService.java

```diff
@@ -314,16 +314,18 @@ public void handleResult(TaskResult taskResult) {
             log.warn("Job {} is finished, ignore result, currentStatus={}", je.getId(), je.getStatus());
             return;
         }
-
-        updateJobScheduleEntity(taskResult);
-        taskResultPublisherExecutor
-                .execute(() -> publisher.publishEvent(new DefaultJobProcessUpdateEvent(taskResult)));
-        if (publisher != null && taskResult.getStatus() != null && taskResult.getStatus().isTerminated()) {
-            taskResultPublisherExecutor.execute(() -> publisher
-                    .publishEvent(new JobTerminateEvent(taskResult.getJobIdentity(), taskResult.getStatus())));
-            if (taskResult.getStatus() == JobStatus.FAILED) {
-                AlarmUtils.alarm(AlarmEventNames.TASK_EXECUTION_FAILED,
-                        MessageFormat.format("Job execution failed, jobId={0}", taskResult.getJobIdentity().getId()));
+        int rows = updateJobScheduleEntity(taskResult);
```

Comment:
add status condition for updateJobScheduleEntity, use nativeQuery

Reply:
yes, I use nativeQuery `updateReportResult` with id and old status to update result